autoemulate/emulators/neural_networks/mlp.py

```diff
@@ -26,32 +29,33 @@ def __init__(
             random_state=random_state,
         )
         modules = []
-        for hidden_size in hidden_sizes:
+        assert hidden_layers >= 1
+        for _ in range(hidden_layers):
             modules.append(nn.Linear(in_features=input_size, out_features=hidden_size))
-            modules.append(nn.ReLU())
+            modules.append(hidden_activation())
             input_size = hidden_size
         modules.append(nn.Linear(in_features=input_size, out_features=output_size))
         self.model = nn.Sequential(*modules)
 
     def get_grid_params(self, search_type: str = "random"):
+        param_space = {
+            "max_epochs": np.arange(10, 110, 10).tolist(),
+            "batch_size": np.arange(2, 128, 2).tolist(),
+            "module__hidden_layers": np.arange(1, 4).tolist(),
+            "module__hidden_size": np.arange(50, 250, 50).tolist(),
+            "module__hidden_activation": [
+                nn.ReLU,
+                nn.Tanh,
+                nn.Sigmoid,
+                nn.GELU,
+            ],
+            "optimizer__weight_decay": (1 / 10 ** np.arange(1, 9)).tolist(),
+        }
```

Comment:
This looks good. One thing I wondered is whether we should search over a few optimizers, in particular `lbfgs` (see [here](https://pytorch.org/docs/stable/generated/torch.optim.LBFGS.html) and [scikit-learn-doc](https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPRegressor.html). When I tested a few smaller datasets with the scikit learn MLP, `lbfgs` tended to be the best optimiser. However, it can become prohibitive for larger datasets (which we won't encounter very much I think).

Reply:
I can add `optimizer: [torch.optim.AdamW, torch.optim.SGD]`. `LBFGS` doesn't support weight decay, we would have to setup conditional search space otherwise.

Reply:
Ok, sounds good. Maybe we can open an issue to add LBFGS eventually.